src/extension.ts

```diff
@@ -228,6 +228,36 @@ export namespace Openshift {
     }
 
     export namespace Explorer {
+        export const login = async function loginCluster(odo: odoctl.Odo, explorer: explorerFactory.OpenShiftExplorer) {
+            if(await odo.requireLogin()) {
+                const clusterIP = await vscode.window.showInputBox({prompt: "Mention the IP address of the cluster", validateInput: (value:string) => {
+                    if (value.trim().length === 0) {
+                        return 'Invalid IP';
+                    }
+                    // add IP validation code
+                }});
+
+                const loginMethod = await vscode.window.showQuickPick(['Credentials', 'Token'], {placeHolder: 'Do you want to login using'});
+                if(loginMethod === "Credentials") {
+                    let username = await vscode.window.showInputBox({prompt: "Mention the username", validateInput: (value:string) => {
+                        if (value.trim().length === 0) {
+                            return 'Invalid username';
+                        }
+                    }});
+                    if (!username) { return; }
+                    let passwd  = await vscode.window.showInputBox({prompt: "Mention the password"});
+                    odo.execute(`oc login ${clusterIP} -u ${username} -p ${passwd}`).then(()=>{
+                        explorer.refresh();
+                    });
+                } else {
+                    // add logic to store token in settings and use it.
```

Comment:
Just run `oc login ${clusterIP}  --token='${token}'` and it will be saved in ~/.kube/config.

Reply:
Updated in the latest commit.